sdk/identity/azure-identity/inc/azure/identity/managed_identity_credential.hpp

```diff
@@ -193,11 +205,21 @@ namespace Azure { namespace Identity {
      *
      * @param clientId Client ID.
      * @param options Options for token retrieval.
+     * @param isProbeEnabled if Azure Instance Metadata Service (IMDS) gets selected as managed
+     * identity source, specifies whether the first request should be a short probe request
+     * (`true`), instead of a normal request with retries and exponential backoff (`false`). Default
+     * is `true`.
+     *
+     * @note When @a isProbeEnabled is `true`, there's a potential that the credential would not
+     * detect IMDS being available on a machine, if the response was not received fast enough. When
+     * `false` and IMDS is not available, credential creation may take tens of seconds until
+     * multiple attempts to get a response from IMDS would fail.
      */
     explicit ManagedIdentityCredential(
         std::string const& clientId = std::string(),
         Azure::Core::Credentials::TokenCredentialOptions const& options
-        = Azure::Core::Credentials::TokenCredentialOptions());
+        = Azure::Core::Credentials::TokenCredentialOptions(),
```

Comment:
Is there really no way of implementing this without changing the public API surface for ManagedIdentityCredential?

Reply:
There is always a way, but I was thinking C# SDK does expose it. Plus, I thought it would be a bit strange if you wanted a MIC with no probe for IMDS, and the only way to make it to work would be to use DAC and set the env var to MIC, but not directly.